src/worker/move-contact-worker.ts

```diff
@@ -0,0 +1,160 @@
+import axios from 'axios';
+import { spawn } from 'child_process'; 
+import { Worker, Job } from 'bullmq';
+
+import Auth from '../lib/authentication';
+import { queuePrivateKey, redisConnection } from '../shared/queue-config';
+import { queueManager } from '../shared/queues';
+
+export interface MoveContactData {
+  parentId: string;
+  contactId: string;
+  sessionToken: string;
+  instanceUrl: string;
+}
+
+export type JobResult = { success: boolean; message: string };
+
+export class MoveContactWorker {
+  private readonly DELAY_IN_MILLIS = 360_000;   // 1 hour
+  private readonly MAX_TIMEOUT = 180_000;       // 30 minutes timeout
+  private readonly MAX_CONCURRENCY = 1;         // Limit concurrency to 1 job at a time
+  private readonly MAX_SENTINEL_BACKLOG = 1000; // ensure we don't take down the server
+  
+  constructor(private queueName: string) {
+    this.initializeWorker();
+  }
+
+  private initializeWorker() {
+    return new Worker(
+      this.queueName, 
+      this.handleJob, 
+      { connection: redisConnection, concurrency: this.MAX_CONCURRENCY }
+    );
+  }
+
+  private handleJob = async (job: Job): Promise<boolean> => {
+    const jobData: MoveContactData = job.data;
+
+    // Ensure server availability
+    if (!(await this.canProcess(jobData))) {
```

Comment:
```suggestion
    if (await this.shouldPostpone(jobData)) {
```

? Slightly easier to read imo